Sources/ScipioKit/Resolver/PackageResolver.swift

```diff
@@ -0,0 +1,544 @@
+    private func resolveDependencyPackage(
+        _ packageName: String
+    ) -> DependencyPackage? {
+        // packageName can be different between `show-dependencies` and `dump-package`, so we try all possible cases
+        if let dependencyPackage = dependencyPackagesByName[packageName] {
+            dependencyPackage
+        } else if let dependencyPackage = dependencyPackagesByID[packageName] {
+            dependencyPackage
+        } else if let dependencyPackage = dependencyPackagesByID[packageName.lowercased()] {
+            dependencyPackage
+        } else {
+            nil
+        }
+    }
+
+    /// Resolve a manifest Target into a ResolvedModule.
+    private func resolve(
+        target: Target,
+        in manifest: PackageManifestKit.Manifest
+    ) async throws -> ResolvedModule {
+        let dependencyPackage = resolveDependencyPackage(for: manifest)
+
+        return try await resolve(
+            target: target,
+            in: manifest,
+            dependencyPackage: dependencyPackage
+        )
+    }
+
+    /// Resolve a manifest Target into a ResolvedModule.
+    private func resolve(
+        target: Target,
+        in manifest: Manifest,
+        dependencyPackage: DependencyPackage
+    ) async throws -> ResolvedModule {
+        guard case let .root(localPackageURL) = manifest.packageKind else {
+            preconditionFailure(
+                """
+                manifest.packageKind must be .root because ManifestLoader.loadManifest is invoked for each dependency,
+                so every loaded manifest is treated as a root package."
+                """
+            )
+        }
+
+        let module = try await ResolvedModule(
+            underlying: target,
+            dependencies: resolve(dependencies: target.dependencies, in: manifest),
+            localPackageURL: localPackageURL,
+            packageID: PackageID(packageKind: manifest.packageKind, packageIdentity: dependencyPackage.identity),
+            resolvedModuleType: resolveModuleType(of: target, dependencyPackage: dependencyPackage)
+        )
+        allModules.insert(module)
+        return module
+    }
+
+    /// Normalizes an optional PackageCondition into an array.
+    private func normalizeConditions(_ condition: PackageCondition?) -> [PackageCondition] {
+        return condition.map { [$0] } ?? []
+    }
+
+    /// Determines the type of a module (e.g., Swift, Clang, binary) for a given target.
+    private func resolveModuleType(
+        of target: Target,
+        dependencyPackage: DependencyPackage
+    ) -> ResolvedModuleType {
+        if let cachedModuleType = cachedModuleType[target] {
+            return cachedModuleType
+        }
+
+        let resolvedModuleType = ModuleTypeResolver(
+            fileSystem: fileSystem,
+            rootPackageDirectory: packageDirectory,
+            target: target,
+            dependencyPackage: dependencyPackage
+        ).resolve()
```

Comment:
Looks like we can hold an instance of ModuleTypeResolver as a property of PackageResolver and pass `target` and `dependencyPackage` arguments to `resolve` method.

```suggestion
        let resolvedModuleType = moduleTypeResolver.resolve(
            target: target,
            dependencyPackage: dependencyPackage
        )
```